src/coredump/coredumpctl.c

```diff
@@ -501,6 +566,10 @@ static int print_list(FILE* file, sd_journal *j, Table *t) {
                 RETRIEVE(d, l, "COREDUMP", coredump);
         }
 
+        r = get_absolute_filename(j, &filename);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get absolute filename: %m");
```

Comment:
You can use something like the following instead:

```
        free_and_replace(filename, (char *) { path_make_absolute(filename, j->prefix) });
        if (!filename)
                return log_oom();
```

Reply:
why the weird `(char*) {` thing?

Reply:
`free_and_replace()` takes the address of its arguments, so it needs to be passed a variable that it can take the address of, not an rvalue, so we can use compound initialization to allocate an anonymous variable inline that `free_and_replace()` can take the address of.

Reply:
Disregard this doesn't actually work

Reply:
I checked if we could move this into free_and_replace() itself but this only works when using a temporary variable so it has to be used at the call site